courses/models_test.py

```diff
@@ -60,120 +77,51 @@ def test_no_run(self):
         """
         assert self.course.get_first_unexpired_run() is None
 
-    def test_only_past(self):
-        """
-        enrollment past and course past
-        """
-        self.create_run(
-            start=self.now-timedelta(weeks=52),
-            end=self.now-timedelta(weeks=45),
-            enr_start=self.now-timedelta(weeks=62),
-            enr_end=self.now-timedelta(weeks=53),
-        )
-        assert self.course.get_first_unexpired_run() is None
-
-    def test_present_enroll_closed(self):
-        """
-        enrollment past, course present
-        """
-        self.create_run(
-            start=self.now-timedelta(weeks=1),
-            end=self.now+timedelta(weeks=2),
-            enr_start=self.now-timedelta(weeks=10),
-            enr_end=self.now-timedelta(weeks=3),
-        )
-        assert self.course.get_first_unexpired_run() is None
-
-    def test_present_enroll_closed_course_future(self):
-        """
-        enrollment past, course future
-        """
-        self.create_run(
-            start=self.now+timedelta(weeks=1),
-            end=self.now+timedelta(weeks=3),
-            enr_start=self.now-timedelta(weeks=10),
-            enr_end=self.now-timedelta(weeks=3),
-        )
-        assert self.course.get_first_unexpired_run() is None
-
-    def test_enroll_none_course_past(self):
-        """
-        enrollment none, course past
-        """
-        self.create_run(
-            start=self.now-timedelta(weeks=10),
-            end=self.now-timedelta(weeks=3),
-        )
-        assert self.course.get_first_unexpired_run() is None
-
-    def test_only_present_forever(self):
-        """
-        enrollment none, course started in the past with no end
-        """
-        course_run = self.create_run(
-            start=self.now-timedelta(weeks=52),
-        )
-        next_run = self.course.get_first_unexpired_run()
-        assert isinstance(next_run, CourseRun)
-        assert next_run.pk == course_run.pk
-
-    def test_enr_present_course_future(self):
-        """
-        enrollment present, course future
-        """
-        course_run = self.create_run(
-            start=self.now+timedelta(weeks=1),
-            end=self.now+timedelta(weeks=4),
-            enr_start=self.now-timedelta(weeks=10),
-            enr_end=self.now+timedelta(weeks=1),
-        )
-        next_run = self.course.get_first_unexpired_run()
-        assert isinstance(next_run, CourseRun)
-        assert next_run.pk == course_run.pk
-
-    def test_present_enroll_open(self):
+    @data(
+        [-52, -45, -62, -53, False],
+        [-1, 2, -10, -3, False],
+        [1, 3, -10, -3, False],
+        [-10, -3, None, None, False],
+        [-52, None, None, None, True],
+        [1, 4, -10, 1, True],
+        [-1, 2, -10, 1, True],
+        [-1, 2, -10, None, True],
+        [52, 62, 40, 50, True],
+        [1, 4, None, None, True],
```

Comment:
I definitively need this otherwise it is way too complicated to match the old tests with these ddt lists